File: src/main/java/com/trustline/trustline/appuser/controller/UserController.java
package com.trustline.trustline.appuser.controller;


import com.trustline.trustline.appuser.dto.*;
import com.trustline.trustline.appuser.service.EmailService;
import com.trustline.trustline.appuser.service.UserService;
import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

@AllArgsConstructor
@RestController
@RequestMapping("api/v1/auth")
public class UserController {

    private final UserService userService;


    @PostMapping("/register")
    public UserResponseDto register(@RequestBody @Validated RegisterUserDto registerUserDto) {
        return UserResponseDto.fromUser(userService.createUser(registerUserDto));
    }

    @PostMapping("/verify-otp")
    public String verifyOtp(
            @Validated @RequestBody OtpRequest otpRequest
    ) {
        return userService.verifyOtp(otpRequest);
    }

    @PostMapping("/verify-password-otp")
    public String verifyPasswordOtp(@Validated @RequestBody OtpRequest otpRequest){
        return userService.resetPasswordOtp(otpRequest);
    }

    @PostMapping("/login")
    public LoginRes<UserResponseDto> login(@Validated @RequestBody LoginReq loginReq) {
        return userService.login(loginReq);
    }

    //    TODO: forgot password to include
    @PostMapping("/forgot-password")
    public UserResponseDto forgotPassword(@Valid @RequestBody ForgotPasswordReq forgotPasswordReq) {
        return UserResponseDto.fromUser(userService.forgotPassword(forgotPasswordReq));
    }

    @PostMapping("/reset-password")
    public UserResponseDto resetPassword(@Valid @RequestBody ResetPasswordReq resetPasswordReq) {
        return UserResponseDto.fromUser(userService.resetPassword(resetPasswordReq));
    }



//  TODO:  forgot password request to generate OTP
//    TODO: after successfully validating OTP, reset password with new password is provided


}

File: src/main/kotlin/trustline/appuser/controller/UserController.kt
package trustline.appuser.controller;


import jakarta.validation.Valid
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import trustline.appuser.dto.*
import trustline.appuser.model.UserResponseDto
import trustline.appuser.service.UserService


@RestController
@RequestMapping("api/v1/auth")
class UserController(
    private val userService: UserService
) {

    @PostMapping("/register")
    fun register(@RequestBody @Validated registerUserDto: RegisterUserDto): UserResponseDto {
        return userService.createUser(registerUserDto)
    }

    @PostMapping("/verify-otp")
    fun verifyOtp(@Validated @RequestBody otpRequest: OtpRequest): OtpVerificationResponse {
        return userService.verifyOtp(otpRequest)
    }

    @PostMapping("/resend-otp")
    fun resendOtp(@Validated @RequestBody resendOtpRequest: ResendOtpRequest): OtpVerificationResponse {
        return userService.resendOtp(resendOtpRequest)
    }

    @PostMapping("/login")
    fun login(@Validated @RequestBody loginReq: LoginReq): LoginRes {
        return userService.login(loginReq)
    }

    @PostMapping("/forgot-password")
    fun forgotPassword(@Valid @RequestBody forgotPasswordReq: ForgotPasswordReq): ForgotPasswordRes {
        return userService.forgotPassword(forgotPasswordReq)
    }

    @PostMapping("/reset-password")
    fun resetPassword(@Valid @RequestBody resetPasswordReq: ResetPasswordReq): UserResponseDto {
        return userService.resetPassword(resetPasswordReq)
    }
}
